pyproject.toml

```diff
@@ -26,6 +26,7 @@ requests = "^2.25.0"
 cryptography = ">=3.4.8"
 tqdm = "^4.32.1"
 pydantic = ">=1.9.0"
+cachetools = "^5.3.3"
```

Comment:
This being the SDK, can we put a lower version of the cache, or use the built-in functools version since we want to maintain a wide range of version compatibiltiy